src/goto-programs/goto_contractor.cpp

```diff
@@ -0,0 +1,359 @@
+//
+// Created by Mohannad Aldughaim on 09/01/2022.
+//
+
+#include "goto_contractor.h"
+
+using namespace std;
+
+
+void goto_contractort::get_constraints(goto_functionst goto_functions)
+{
+  auto function = goto_functions.function_map.find("c:@F@main");
+  auto it = function->second.body.instructions.begin();
+  while(it != function->second.body.instructions.end())
+  {
+    if(it->is_assert())
+      constraint = create_constraint_from_expr2t(it->guard);
+
+    it++;
+  }
+}
+
+void goto_contractort::get_intervals(goto_functionst goto_functions)
+{
+  //only main here
+  auto function = goto_functions.function_map.find("c:@F@main");
+  std::cout << "\n parsing assumes\nfound main: " << function->first << "\n";
+  auto it = function->second.body.instructions.begin();
+
+  while(it != function->second.body.instructions.end())
+  {
+    if(it->is_assume())
+      parse_intervals(it->guard);
+
+    it++;
+  }
+}
+
+void goto_contractort::parse_intervals(irep_container<expr2t> expr)
+{
+  symbol2tc symbol;
+  double value;
+
+  if(!is_comp_expr(expr))
+    return;
+
+  std::shared_ptr<relation_data> rel;
+  rel = dynamic_pointer_cast<relation_data>(expr);
+
+  //expr->dump();
```

Comment:
remove unnecessary comments.

Reply:
done. Though I left some for reference later. Will be removed when I don't need them anymore.